ballerina/websocket_service.bal

```diff
@@ -57,19 +57,17 @@ isolated service class WsService {
             return self.handleError(caller, wsText);
         }
 
-        WSPayload|json|error wsPayload = self.customHeaders != {}
-                                        ? wsText.cloneWithType(WSPayload) : value:fromJsonString(text);
-        if wsPayload is error {
-            return self.handleError(caller, wsPayload);
+        WSPayload|json|error payload = self.customHeaders != {}
+            ? wsText.cloneWithType(WSPayload) : value:fromJsonString(text);
+        if payload is error {
+            return self.handleError(caller, payload);
         }
-        string wsType = wsPayload is WSPayload ? <string>wsPayload.'type : DEFAULT_VALUE;
-        string connectionId = wsPayload is WSPayload && wsPayload?.id !is () ? <string>wsPayload?.id : DEFAULT_VALUE;
-
         if !self.customHeaders.hasKey(WS_SUB_PROTOCOL) {
-            return self.handleSubscriptionRequest(caller, connectionId, wsPayload);
+            return self.handleSubscriptionRequest(caller, DEFAULT_VALUE, payload);
         }
 
-        match wsType {
+        WSPayload wsPayload = <WSPayload>payload;
```

Comment:
Why do we need this casting here?

Reply:
In line 65 we have check for `!self.customHeaders.hasKey(WS_SUB_PROTOCOL)` instead of checking `payload is json` this makes the resulting type to have `WSPayload|json`. Casting is done to eliminate json type from the union.

We also could remove the casting by checking `payload is json` in line 65. but this could make code slightly unclear

Reply:
We can type-lift the json type as well then.

Reply:
WSPayload|json is ambiguous type type. 

the other way we could avoid casting is to replace the casting with following if block
```
if wsPayload !is WSPayload {
   return;
}
```
will update the PR with this

Reply:
updated